### ISSUE TITLE
Validator without ACKs ?

### ISSUE THREAD

**Issue by SurfingNerd** [DMDcoin/openethereum-3.x#65]
We found evidence in cases where validators can join a validator set, without writing acks.

```
example:
epoch switch:   3035
Block:  93504
Validator: 0x396Aa74164dc9bfF256f994C725155DD456fCC43
Did send his ACKs but those got never included in a block.
 http://explorer.uniq.diamonds/tx/0xc56cf78664d550476ce631a1d4c22e016a1e9b8a50e4e6835207c9ad7f98c2a9
 the transaction got replaced (nonce rule) later by this transaction:
 http://explorer.uniq.diamonds/tx/0xb5d8c525a2e9c398bb775a79e86a13b8acae137c157b509c196f534b6b495099/internal-transactions
```

Is it possible that those ACKS are still required for creating valid HBBFT messages ?
Maybe connected to https://github.com/DMDcoin/openethereum-3.x/issues/64 ?

**Comment by SurfingNerd** [comment]
in progress: Debug printing the count of parts and acks in https://github.com/SurfingNerd/openethereum-3.x/commit/7242e2f6061d1173229d703a36b37253c5d20407

**Comment by SurfingNerd** [comment]
observed another case:
epoch switch at 104450
ACK tx 0x3e4f45027c924f9b4a803162cee14f205a35ea4fc460a5fa3a28884f6f1b577f  from 0xF001FaaA1aee5Cc972838766304398B97e909B63  could not get included.
despite that, the node got added to the validator set, but was kicked on epoch later.

http://explorer.uniq.diamonds/blocks/104450/transactions

**Comment by SurfingNerd** [comment]
Fixed with [PR73](https://github.com/DMDcoin/openethereum-3.x/pull/73)
Released with [v3.2.5-hbbft-0.8.0](https://github.com/DMDcoin/openethereum-3.x/releases/tag/v3.2.5-hbbft-0.8.0)
